### PR TITLE
fix(components): [date-picker] support dynamic setting of `format`

### LINKED ISSUE #17149: [Component] [date-picker] DatePicker format动态修改无效
Bug Type: **`Component`**

## Environment

- Vue Version: `3.4.0`
- Element Plus Version: `2.7.4`
- Browser / OS: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36`
- Build Tool: `Vite`

### Related Component

- `el-date-picker`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGRpdiBjbGFzcz1cImRlbW8tZGF0ZXRpbWUtcGlja2VyXCI+XG4gICAgPGRpdiBjbGFzcz1cImJsb2NrXCI+XG4gICAgICA8c3BhbiBjbGFzcz1cImRlbW9uc3RyYXRpb25cIj5FbWl0cyBEYXRlIG9iamVjdDwvc3Bhbj5cbiAgICAgIDxkaXYgY2xhc3M9XCJkZW1vbnN0cmF0aW9uXCI+VmFsdWU6IHt7IHZhbHVlMSB9fTwvZGl2PlxuICAgICAgPGVsLWRhdGUtcGlja2VyXG4gICAgICAgIHYtbW9kZWw9XCJ2YWx1ZTFcIlxuICAgICAgICB0eXBlPVwiZGF0ZXRpbWVyYW5nZVwiXG4gICAgICAgIHBsYWNlaG9sZGVyPVwiUGljayBhIERhdGVcIlxuICAgICAgICA6Zm9ybWF0PVwiZm9ybWF0MlwiXG4gICAgICAvPlxuICAgIDwvZGl2PlxuICA8L2Rpdj5cbiAgPGVsLWJ1dHRvbiBAY2xpY2s9XCJjaGFuZ2VcIj5mb3JtYXTkv67mlLk8L2VsLWJ1dHRvbj5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYscmVhY3RpdmUgfSBmcm9tICd2dWUnXG5cbmNvbnN0IHZhbHVlMSA9IHJlZignJylcbmNvbnN0IHZhbHVlMiA9IHJlZignJylcbmNvbnN0IHZhbHVlMyA9IHJlZignJylcbmNvbnN0IGZvcm1hdDIgPSByZWYoXCJZWVlZL01NL0REIEhIOm1tOnNzXCIpXG5jb25zdCBjaGFuZ2UgPSAoKSA9PiB7XG4gIGZvcm1hdDIudmFsdWUgPSBcIllZWVktTU0tRERcIlxufVxuPC9zY3JpcHQ+XG48c3R5bGUgc2NvcGVkPlxuLmRlbW8tZGF0ZXRpbWUtcGlja2VyIHtcbiAgZGlzcGxheTogZmxleDtcbiAgd2lkdGg6IDEwMCU7XG4gIHBhZGRpbmc6IDA7XG4gIGZsZXgtd3JhcDogd3JhcDtcbn1cbi5kZW1vLWRhdGV0aW1lLXBpY2tlciAuYmxvY2sge1xuICBwYWRkaW5nOiAzMHB4IDA7XG4gIHRleHQtYWxpZ246IGNlbnRlcjtcbiAgYm9yZGVyLXJpZ2h0OiBzb2xpZCAxcHggdmFyKC0tZWwtYm9yZGVyLWNvbG9yKTtcbiAgZmxleDogMTtcbn1cbi5kZW1vLWRhdGV0aW1lLXBpY2tlciAuYmxvY2s6bGFzdC1jaGlsZCB7XG4gIGJvcmRlci1yaWdodDogbm9uZTtcbn1cbi5kZW1vLWRhdGV0aW1lLXBpY2tlciAuZGVtb25zdHJhdGlvbiB7XG4gIGRpc3BsYXk6IGJsb2NrO1xuICBjb2xvcjogdmFyKC0tZWwtdGV4dC1jb2xvci1zZWNvbmRhcnkpO1xuICBmb250LXNpemU6IDE0cHg7XG4gIG1hcmdpbi1ib3R0b206IDIwcHg7XG59XG48L3N0eWxlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiX28iOnt9fQ==)

### Steps to reproduce

点击按钮之后 选择日期

### What is Expected?

点击按钮之后 选择日期根据格式展示日期
或者 修改change方法 根据return出来的值进行展示 

### What is actually happening?

点击按钮之后 选择日期依旧是之前的format格式

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by dddssw**
change format will not update component

closed #17149

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Review by polikm5 — COMMENTED** [review]
试了一下 确实是这里的问题

**Comment by dddssw** [comment]
emm,there is a error ?

**Comment by warmthsea** [comment]
> emm,there is a error ?

You don't need to deal with it, it's not caused by your code.

**Comment by dddssw** [comment]
all right , thank you @warmthsea

**Review by btea — COMMENTED** [review]
Can you add a test case?

**Comment by dddssw** [comment]
Sorry,I don’t know how to add a test case. It would be better if there are articles to guide。 @btea

**Comment by btea** [comment]
You can refer to existing test cases.
https://github.com/element-plus/element-plus/blob/dev/packages/components/date-picker/__tests__/date-picker.test.ts

**Comment by dddssw** [comment]
I wil give a try

**Comment by dddssw** [comment]
Could you please review the test case I wrote? @btea

File: packages/components/date-picker/__tests__/date-picker.test.ts [hunk 1]
```diff
@@ -1633,4 +1633,30 @@ describe('MonthRange', () => {
...
+          changeFormat() {
+            this.format = 'YYYY-MM-DD'
+          },
+        },
+      }
+    )
+    await wrapper.find('button').trigger('click')
+    await nextTick()
+    const vm = wrapper.vm as any
+    expect(vm.format).toBe('YYYY-MM-DD')
```
Comment (btea) [inline packages/components/date-picker/__tests__/date-picker.test.ts:1660]: This test case should verify the change in value before and after the `format` change. The following are the adjustments I made based on your code.

image

File: packages/components/date-picker/__tests__/date-picker.test.ts [hunk 2]
```diff
@@ -1633,4 +1633,30 @@ describe('MonthRange', () => {
     expect(vm.value.getDate()).toBe(1)
     expect(vm.value.getHours()).toBe(12)
   })
+  it('prop format should be Responsive for monthRange', async () => {
```
Comment (btea) [inline packages/components/date-picker/__tests__/date-picker.test.ts:1636]: The description can be changed to `format allows dynamic changes`.

**Comment by dddssw** [comment]
Thanks for your review

**Review by btea — APPROVED** [review]
Thanks!

**Review by FrontEndDog — APPROVED** [review]
Thanks for your contribution！

**Pull request merged**